polaris-service/src/main/java/org/apache/polaris/service/PolarisApplication.java

```diff
@@ -412,4 +533,31 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
       }
     }
   }
+
+  private static class PolarisMetaStoreManagerFactory implements Factory<PolarisMetaStoreManager> {
```

Comment:
Should be replaced with `@Produces` and `@Disposes`

Reply:
Mentioned above - HK2 doesn't support `@Produces` out of the box. HK2 implements "CDI-lite" whereas the enterprise.* annotations are defined in the "CDI-full" spec